tests/system/test_cluster/test_agent_groups/test_assign_groups_guess.py

```diff
@@ -0,0 +1,161 @@
+    assertions:
+        - Verify that after registering the agent key file exists in all nodes.
+        - Verify that after registering the agent appears as never_connected in all nodes.
+        - Verify that after registering it has the 'Null' group assigned.
+        - Verify that after assign group with agent-groups the change is sync with the cluster.
+    expected_output:
+        - The agent 'Agent_name' with ID 'Agent_id' belongs to groups: group_test."
+    '''
+    group_id = 'group_test'
+    # Modify local_internal_options
+    replace = '\n' + remoted_guess_agent_groups + f'{status_guess_agent_group}\n'
+
+    if(int(status_guess_agent_group) == 1):
+        for host in test_infra_managers:
+            host_manager.add_block_to_file(host, path=f"{WAZUH_PATH}/etc/local_internal_options.conf",
+                                           after="upgrades.", before="authd.debug=2", replace=replace)
+        # Restart managers
+        restart_cluster(test_infra_managers, host_manager)
+        time.sleep(timeout)
+
+    # Register agent with agent-auth
+    agent_ip, agent_id, agent_name, manager_ip = register_agent(test_infra_agents[0], agent_target,
+                                                                host_manager)
+    # Restart agent
+    restart_cluster(test_infra_agents, host_manager)
+
+    # Check that agent has client key file
+    assert check_keys_file(test_infra_agents[0], host_manager)
+
+    try:
+        # Create new group and assing agent
+        assign_agent_to_new_group(test_infra_managers[0], group_id, agent_id, host_manager)
+
+        time.sleep(timeout)
+
+        # Check that agent has group set to group_test on Managers
+        check_agent_groups(agent_id, group_id, test_infra_managers, host_manager)
+
+        check_agent_status(agent_id, agent_name, agent_ip, AGENT_STATUS_ACTIVE, host_manager, test_infra_managers)
+
+        # Remove the agent
+        remove_cluster_agents(test_infra_managers[0], test_infra_agents, host_manager)
+
+        # Register agent with agent-auth
+        agent_ip, agent_id, agent_name, manager_ip = register_agent(test_infra_agents[0], agent_target,
+                                                                    host_manager)
+
+        # Check that agent has client key file
+        assert check_keys_file(test_infra_agents[0], host_manager)
+
+        # Restart agent
+        restart_cluster(test_infra_agents, host_manager)
+        time.sleep(timeout)
+        check_agent_status(agent_id, agent_name, agent_ip, AGENT_STATUS_ACTIVE, host_manager, test_infra_managers)
+
+        # Check if remoted.guess_agent_group is disabled
+        if(int(status_guess_agent_group) == 0):
+            group_id = 'default'
+        check_agent_groups(agent_id, group_id, test_infra_managers, host_manager)
+
+    finally:
+        # Delete group of agent
+        delete_group_of_agents(test_infra_managers[0], group_id, host_manager)
```

Comment:
Is there any reason, because it is on `try..finally` section? 
The clean_environment method shouldn't remove all.